tests/unit/test_range.cpp

```diff
@@ -305,3 +306,138 @@ TEST_F(RangeTest, StringContainsValue) {
 
 	StringRange_Free(r);
 }
+
+//------------------------------------------------------------------------------
+// Unsigned range
+//------------------------------------------------------------------------------
+
+TEST_F(RangeTest, UnsignedRangeNew) {
+	UnsignedRange *r = UnsignedRange_New();
+
+	ASSERT_TRUE(r->valid);
+	ASSERT_EQ(r->max, UINT64_MAX);
+	ASSERT_EQ(r->min, 0);
+	ASSERT_TRUE(r->include_max);
+	ASSERT_TRUE(r->include_min);
+
+	UnsignedRange_Free(r);
+}
+
+TEST_F(RangeTest, UnsignedRangeValidation) {
+	UnsignedRange *r = UnsignedRange_New();
+
+	// X > 5 AND X < 5
+	r->max = 5;
+	r->min = 5;
+	r->include_max = false;
+	r->include_min = false;
+	ASSERT_FALSE(UnsignedRange_IsValid(r));
+
+	// X >= 5 AND X < 5
+	r->max = 5;
+	r->min = 5;
+	r->include_max = false;
+	r->include_min = true;
+	ASSERT_FALSE(UnsignedRange_IsValid(r));
+
+	// X >= 5 AND X <= 5
+	r->max = 5;
+	r->min = 5;
+	r->include_max = true;
+	r->include_min = true;
+	ASSERT_TRUE(UnsignedRange_IsValid(r));
+
+	// X > 5 AND X <= 5
+	r->max = 5;
+	r->min = 5;
+	r->include_max = true;
+	r->include_min = false;
+	ASSERT_FALSE(UnsignedRange_IsValid(r));
+
+	// (5, 10)  X > 5 AND x < 10.
+	r->max = 10;
+	r->min = 5;
+	r->include_max = false;
+	r->include_min = false;
+	ASSERT_TRUE(UnsignedRange_IsValid(r));
+
+	// (5, 10]  X > 5 AND x <= 10.
+	r->max = 10;
+	r->min = 5;
+	r->include_max = true;
+	r->include_min = false;
+	ASSERT_TRUE(UnsignedRange_IsValid(r));
+
+	// [5, 10)  X >= 5 AND x < 10.
+	r->max = 10;
+	r->min = 5;
+	r->include_max = false;
+	r->include_min = true;
+	ASSERT_TRUE(UnsignedRange_IsValid(r));
+
+	// [5, 10]  X >= 5 AND x =< 10.
+	r->max = 10;
+	r->min = 5;
+	r->include_max = true;
+	r->include_min = true;
+	ASSERT_TRUE(UnsignedRange_IsValid(r));
+
+	UnsignedRange_Free(r);
+}
+
+TEST_F(RangeTest, UnsignedTightenRange) {
+	UnsignedRange *r = UnsignedRange_New();
+
+	// X < 100
+	UnsignedRange_TightenRange(r, OP_LT, 100);
+	ASSERT_EQ(r->max, 100);
+	ASSERT_FALSE(r->include_max);
+
+	// X <= 100
+	UnsignedRange_TightenRange(r, OP_LE, 100);
+	ASSERT_EQ(r->max, 100);
+	ASSERT_FALSE(r->include_max);
```

Comment:
This test checks for tightening the range. It starts with op LT, which sets `include_max` to false. Since you can't tighten LT with LE, it remains false.